pkg/objstore/oss/oss.go

```diff
@@ -0,0 +1,289 @@
+func (b *Bucket) Delete(ctx context.Context, name string) error {
+	if err := b.bucket.DeleteObject(name); err != nil {
+		return errors.Wrap(err, "delete oss object")
+	}
+	return nil
+}
+
+func configFromEnv() Config {
+	c := Config{
+		Endpoint:        os.Getenv("ALIYUNOSS_ENDPOINT"),
+		Bucket:          os.Getenv("ALIYUNOSS_BUCKET"),
+		AccessKeyID:     os.Getenv("ALIYUNOSS_ACCESS_KEY_ID"),
+		AccessKeySecret: os.Getenv("ALIYUNOSS_ACCESS_KEY_SECRET"),
+	}
+	return c
+}
+
+func NewBucket(logger log.Logger, conf []byte, component string) (*Bucket, error) {
+	config, err := func(conf []byte) (Config, error) {
+		var config Config
+		if err := yaml.Unmarshal(conf, &config); err != nil {
+			return Config{}, err
+		}
+
+		return config, nil
+	} (conf)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "parse aliyun oss config file failed")
+	}
+	if err := validate(config); err != nil {
+		return nil, errors.Wrap(err, "validate aliyun oss config file failed")
+	}
+
+	client, err := alioss.New(config.Endpoint, config.AccessKeyID, config.AccessKeySecret)
+	if err != nil {
+		return nil, errors.Wrap(err, "create aliyun oss client failed")
+	}
+	bk, err := client.Bucket(config.Bucket)
+	if err != nil {
+		return nil, errors.Wrap(err, "use oss bucket "+ config.Bucket+" failed")
+	}
+
+	bkt := &Bucket{
+		logger: logger,
+		client: client,
+		name:   config.Bucket,
+		config: config,
+		bucket: bk,
+	}
+	return bkt, nil
+}
+
+// Iter calls f for each entry in the given directory (not recursive). The argument to f is the full
+// object name including the prefix of the inspected directory.
+func (b *Bucket) Iter(ctx context.Context, dir string, f func(string) error) error {
+	if dir != "" {
+		dir = strings.TrimSuffix(dir, objstore.DirDelim) + objstore.DirDelim
+	}
+
+	marker := alioss.Marker("")
+	for {
+		if err := ctx.Err() ; err != nil {
+			return  errors.Wrap(err, "a done channel closed with errors")
+		}
+		objects, err := b.bucket.ListObjects(alioss.Prefix(dir), alioss.Delimiter(objstore.DirDelim), marker)
+		if err != nil {
+			return errors.Wrap(err, "oss bucket list")
+		}
+		marker = alioss.Marker(objects.NextMarker)
+
+		for _, object := range objects.Objects {
+			if err := f(object.Key); err != nil {
+				return errors.Wrap(err, "callback func invoke with filename "+object.Key+" failed")
+			}
+		}
+
+		for _, object := range objects.CommonPrefixes {
+			if err := f(object); err != nil {
+				return errors.Wrap(err, "callback func invoke with dirname "+object+" failed")
+			}
+		}
+		if !objects.IsTruncated {
+			break
+		}
+	}
+
+	return nil
+}
+
+func (b *Bucket) Name() string {
+	return b.name
+}
+
+// validate checks to see the config options are set.
+func validate(conf Config) error {
+	if conf.Endpoint == "" {
+		return errors.New("no oss endpoint in config file")
+	}
+	if conf.Bucket == "" {
+		return errors.New("no oss bucket in config file")
+	}
+
+	if conf.AccessKeyID == "" || conf.AccessKeySecret == "" {
```

Comment:
I think we should split it and change the error messages to:
```
return errors.New("access_key_id is not present in config file")
```

and
```
return errors.New("access_key_secret is not present in config file")
```

or something similiar